File: google_calendar_api.py
from google.oauth2 import service_account
from googleapiclient.discovery import build
import os

SCOPES = ['https://www.googleapis.com/auth/calendar']

class GoogleCalendarAPI:
    def __init__(self):
        self.service = self.get_calendar_service()

    def get_calendar_service(self):
        credentials = service_account.Credentials.from_service_account_file(
            'service_account.json', scopes=SCOPES)
        return build('calendar', 'v3', credentials=credentials)

    async def create_event(self, event_details):
        event = self.service.events().insert(calendarId='primary', body=event_details).execute()
        return event

    async def list_events(self, time_range):
        events_result = self.service.events().list(calendarId='primary', timeMin=time_range['start'],
                                                   timeMax=time_range['end'], singleEvents=True,
                                                   orderBy='startTime').execute()
        return events_result.get('items', [])

    async def update_event(self, event_id, updates):
        event = self.service.events().get(calendarId='primary', eventId=event_id).execute()
        for key, value in updates.items():
            event[key] = value
        updated_event = self.service.events().update(calendarId='primary', eventId=event_id, body=event).execute()
        return updated_event

    async def delete_event(self, event_id):
        self.service.events().delete(calendarId='primary', eventId=event_id).execute()

File: google_calendar_api.js
const { google } = require('googleapis');
const fs = require('fs');

class GoogleCalendarAPI {
  constructor() {
    this.auth = this.getAuth();
    this.calendar = google.calendar({ version: 'v3', auth: this.auth });
  }

  getAuth() {
    const credentials = JSON.parse(fs.readFileSync('credentials.json'));
    return new google.auth.JWT(
      credentials.client_email,
      null,
      credentials.private_key,
      ['https://www.googleapis.com/auth/calendar']
    );
  }

  async createEvent(eventDetails) {
    const response = await this.calendar.events.insert({
      calendarId: 'primary',
      resource: eventDetails,
    });
    return response.data;
  }

  async listEvents(timeRange) {
    const response = await this.calendar.events.list({
      calendarId: 'primary',
      timeMin: timeRange.start,
      timeMax: timeRange.end,
      singleEvents: true,
      orderBy: 'startTime',
    });
    return response.data.items;
  }

  async updateEvent(eventId, updates) {
    const response = await this.calendar.events.patch({
      calendarId: 'primary',
      eventId: eventId,
      resource: updates,
    });
    return response.data;
  }

  async deleteEvent(eventId) {
    await this.calendar.events.delete({
      calendarId: 'primary',
      eventId: eventId,
    });
  }
}

module.exports = GoogleCalendarAPI;